Add "Exception and log message guideline" section to README (#1054)
scalar-labs/scalardb: README.md

The code formatter is automatically executed when commiting files. A commit will fail and be formatted by the formatter when any invalid code format is detected. Try to commit the change again.

> ### Exception and log message guideline

Reviewer: ```suggestion
### Exception and log message guidelines
```

> All the exception and log messages in this project are consistent with the following guideline.
> 
> - The first character is capitalized
> - The message should not end with punctuation
> 
> Follow the guideline when you contribute to this project.

Reviewer: ```suggestion
All the exception and log messages in this project are consistent with the following guidelines:

- The first character is capitalized.
- The message does not end with a punctuation mark.

When contributing to this project, please follow those guidelines.
```

The code formatter is automatically executed when commiting files. A commit will fail and be formatted by the formatter when any invalid code format is detected. Try to commit the change again.

> ### Exception and log message guideline

Reviewer 2: I think this is a good thing to note.
But I'm wondering if there is any better place to put this section since this might be a bit too specific to put it in the top README.
If it's mainly for contributors, it can be in the `Contributing` session.

@josh-wong Any idea?

Author: > But I'm wondering if there is any better place to put this section since this might be a bit too specific to put it in the top README.

This subsection is already under `Development` not the top level.

Which did you mean
- moving `Development` section under `Contributing`
- or moving the subsections `Pre-commit hook` and `Exception and log message guideline` under `Contributing`?
  - then we can remove `Development` section

Reviewer: > But I'm wondering if there is any better place to put this section since this might be a bit too specific to put it in the top README.
> 
> This subsection is already under `Development` not the top level.
> 
> Which did you mean
> 
> * moving `Development` section under `Contributing`
> * or moving the subsections `Pre-commit hook` and `Exception and log message guideline` under `Contributing`?
>   
>   * then we can remove `Development` section

I agree with @komamitsu's second bullet point.

Since the `Development` heading isn't followed by any text, I think we can remove that heading. 

In that case, the structure would look fine as follows:

- `Contributing`
  - `Pre-commit hook`
  - `Exception and log message guideline`

Reviewer 2: @komamitsu @josh-wong 
Sorry, I misunderstood the `Exception and log message guideline` section was not in the top level.
The structure Josh recommended looks good to me!

Author: @josh-wong @feeblefakie Thanks for the comments! Updated the structure in https://github.com/scalar-labs/scalardb/pull/1054/commits/82615e0a7a339459858c0c6814e18d2e3425a4cd (cc: @brfrn169)
